include/deal.II/lac/computation.h

```diff
@@ -0,0 +1,639 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2010 - 2015 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef __deal2__computation_h
+#define __deal2__computation_h
+
+#include <deal.II/base/config.h>
+#include <deal.II/base/exceptions.h>
+#include <deal.II/lac/vector_memory.h>
+
+#ifdef DEAL_II_WITH_CXX11
+
+DEAL_II_NAMESPACE_OPEN
+
+// Forward declarations:
+
+template <typename Range, typename Domain> class LinearOperator;
+template <typename Range = Vector<double> > class Computation;
+
+
+/**
+ * A class to store a computation.
+ *
+ * The class essentially consists of <code>std::function</code> objects
+ * that store the knowledge of how to generate the result of a computation:
```

Comment:
I think this sounds just overly generic. You specifically intend this to be for operations on vectors, right? I know that it is in general possible to use this for other types as well, but maybe being more specific/concrete helps readers of the documentation. I think it may also be worthwhile considering a more descriptive name for the class. `Operator` or `NonlinearOperator` come to mind. You could also be inspired by C++11's `std::packaged_task`: maybe `PackagedOperation`?